<launch>
  <node name="uts_recogniser" cwd="node" pkg="uts_recogniser" type="uts_recogniser" output="screen">

    <!-- json file name -->
    <param name="json" type="str" value="../data/201505010645.json"/>

    <!-- method file name -->
    <param name="method" type="str" value="../data/method.txt"/>

    <!-- kernel descriptor parameter directory -->
    <param name="kd_dir" type="str" value="../data/kd_models/"/>

    <!-- eblearn dataset directory -->
    <param name="eb_dir" type="str" value="../data/eblearn/models/"/>
    <param name="eb_temp_conf" type="str" value="../data/eblearn/template.conf"/>

    <!-- xml dataset directory -->
    <param name="xml_dir" type="str" value="../data/amazon_models/"/>

    <!-- mask image for each bin directory -->
    <param name="mask_dir" type="str" value="../data/201505010645/"/>

    <!-- use point cloud or depth image -->
    <param name="use_cloud" type="bool" value="false"/>

    <!-- operation mode -->
    <!-- op_mode_ = 1, kernel descriptor only -->
    <!-- op_mode_ = 2, eblearn descriptor only -->
    <!-- op_mode_ = 3, kernel descriptor + eblearn -->
    <!-- op_mode_ = 4, kernel descriptor + eblearn + optional rgb/rgbd -->
    <param name="op_mode" type="int" value="2"/>

    <!-- service mode -->
    <!-- srv_mode_ = 1, non-blocking mode -->
    <!-- srv_mode_ = 2, blocking mode -->
    <param name="srv_mode" type="int" value="2"/>

    <!-- topic names -->
    <param name="xtion_rgb_image" type="string" value="/camera/lowres_rgb/image"/>
    <param name="xtion_depth_image" type="string" value="/camera/depth/image"/>
    <param name="xtion_rgb_info" type="string" value="/camera/lowres_rgb/camera_info"/>
    <param name="xtion_depth_info" type="string" value="/camera/depth/camera_info"/>
    <param name="pg_rgb_image" type="string" value="/camera/highres_rgb/image"/>
    <param name="pg_rgb_info" type="string" value="/camera/highres_rgb/camera_info"/>
    <param name="xtion_points" type="string" value="/camera/points"/>

  </node>
</launch>
